HW6.py

```diff
@@ -0,0 +1,224 @@
+                product.in_freeze_cam = None
+            elif product.in_cold_cam == refr:
+                refr.cold_cells -= product.size
+                product.in_cold_cam = None
+            else:
+                print('{0} is not in {1}.{2} still hungry'.format(product.name,
+                                                                  refr.name,
+                                                                  self.name))
+
+    def put_product(self, refr, product):
+        """Method try to put product in refrigerator
+
+        If refrigerator is broken or product is already in refrigerator, we do
+        nothing.
+
+        Otherwise if it is "cold", we tried to find place for it in cold camera
+        If it it's not enough place in cold camera or it is freeze product,
+        we tried to put product in freeze camera.
+        If it it's not enough place in freeze camera, we will not put product
+        in refrigerator.
+        """
+        if refr.broken:
+            print('{} is broken!'.format(refr.name))
+        elif product.in_cold_cam == refr or product.in_freeze_cam == refr:
+            print('{} is already in {}!'.format(product.name, refr.name))
+        else:
+            if product.pr_type == 'cold' and (refr.cold_cells + product.size <=
+                                              refr.cold_cam_capacity):
+                product.in_cold_cam = refr
+                refr.cold_cells += product.size
+            elif refr.freeze_cells + product.size <= refr.freeze_cam_capacity:
+                product.in_freeze_cam = refr
+                refr.freeze_cells += product.size
+            else:
+                print('{0} can\'t put {1} in {2} - not enough place!'
+                      .format(self.name, product.name, refr.name))
+
+    def eat(self, product):
+        """If product is in refrigerator, in will be deleted forever"""
+        if not product.in_freeze_cam or not product.in_cold_cam:
+            print('{} can\'t eat product from refrigerator!'.format(self.name))
+        else:
+            print('{} eat {}'.format(self.name, product.name))
+            del product
+
+
+class Parent(User):
+    """Parent is User who can call to master, pay and earn money"""
+    money = 1400
+
+    def call_to_master(self, master, refr):
+        """Using this method parent can call to master.
+
+        After call master will try to fix refrigerator using master.try_to_fix
+        method. This method returns result (True o False), and this result will
+        be printed
+        """
+        job_result = master.try_to_fix(self, refr)
+        print('{} called to {}.'.format(self.name, master.name), end=' ')
+        print('He fixed it' if job_result else 'He can\'t fix it')
+
+    def earn_money(self, salary=200):
+        """Parent get some money after this method"""
+        self.money += salary
+
+
+class Child(User):
+    """Child is user who can hit refrigerator"""
+    def hit_fridge(self, refr):
```

Comment:
maybe it should be added to get or put food method for child?

Reply:
But child have "get" and "put" methods, because class Child is inheritor of class User, and class User have those methods

Reply:
we can reload them, call super and add private _hit method to them I meant